docs/src/main/sphinx/routines/function.md

```diff
@@ -0,0 +1,92 @@
+# FUNCTION
+
+## Synopsis
+
+```text
+FUNCTION name [ (parameter1, parameter2, ...) ]
+  RETURNS type
+  [ LANGUAGE langaugeIdentifier]
+  [ NOT? DETERMINISTIC ]
+  [ RETURNS NULL ON NULL INPUT ]
+  [ CALLED ON NULL INPUT ]
+  [ SECURITY { DEFINER | INVOKER } ]
+  [ COMMENT description]
+  statements
+```
+
+## Description
+
+Declare a SQL routine.
+
+The `name` must qualify the name of the catalog to store the routine, unless the
+function is only declared in the scope of a query.
+
+The optional list of parameters is a comma-separated list of names and
+data types.
+
+The `type` value after the `RETURNS` keyword identifies the [data type]() of the
+routine output.
+
+The optional `LANGUAGE` statement identifies the language used for the routine
+definition with `languageIdentifier`. Only `SQL` is supported.
+
+The optional `DETERMINISTIC` or `NOT DETERMINISTIC` statement declared to the
+query engine, if repeated functions calls with identical input parameters yield
+the same result.
+
+The optional `RETURNS NULL ON NULL INPUT` is used to declare that the function
+returns a `NULL` value when `NULL` values are provided as input. This
```

Comment:
Let's change the second sentence, since this is a guarantee

> The function will not be invoked with a `NULL` input value.

Reply:
Done